### PR TITLE
New Partner APIs for ordering service

### PR DESCRIPTION
**Pull request by Pramila-dhaka**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json [hunk 1]
```diff
@@ -0,0 +1,899 @@
...
+            "schema": {
+              "$ref": "#/definitions/SearchInventoriesRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/PartnerInventoryList"
```
Comment (pilor) [inline specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json:212]: This should be x-ms-pageable with a nextLink as I imagine inventories can/will get quite large (especially with each item having a fairly large body)

File: specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/examples/ManageInventoryMetadata.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      "https://localhost/subscriptions/fa68082f-8ff7-4a25-95c7-ce9da541242f/providers/Microsoft.EdgeOrderPartner/locations/westus/inventories/SerialNumber1/manageInventoryMetadata?api-version=2020-12-01-preview"
+    ],
+    "Content-Type": [
+      "application/json; charset=utf-8"
+    ],
+    "Content-Length": [
+      "181"
+    ],
+    "manageInventoryMetadataRequest": {
+      "familyIdentifier": "FamilyIdentifier",
```
Comment (pilor) [inline specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/examples/ManageInventoryMetadata.json:20]: Examples should contain realistic values please

File: specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json (hunk 1 above)
Comment (Pramila-dhaka) [reply specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json:218]: This is fixed.

File: specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/examples/ManageInventoryMetadata.json (hunk 2 above)
Comment (Pramila-dhaka) [reply specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/examples/ManageInventoryMetadata.json]: This is fixed.

File: specification/edgeorderpartner/resource-manager/readme.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Configuration
+
+
+
+### Basic Information
+These are the global settings for the EdgeOrderPartner API.
+
+``` yaml
+title: Edge Order Partner API's
```
Comment (weidongxu-microsoft) [inline specification/edgeorderpartner/resource-manager/readme.md:28]: It is recommended to take a SDK friendly `title`. It is usually used as entry point of the SDK. It usually cannot display space or quote.

E.g. compute https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/readme.md#basic-information
results in SDK class https://docs.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.computemanagementclient?view=azure-python

You can put the detail in `description`.

File: specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json [hunk 4]
```diff
@@ -0,0 +1,905 @@
...
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.EdgeOrderPartner/locations/{location}/productFamilies/{familyIdentifier}/inventories/{serialNumber}/manageLink": {
+      "post": {
+        "tags": [
+          "PartnerInventory"
+        ],
+        "summary": "API for linking management resource with inventory",
+        "operationId": "ManageLink",
```
Comment (weidongxu-microsoft) [inline specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json:130]: Usually it is recommended to write it as `<Group>_<Operation>`, especially if there is pending more operations to be added. You can refer to json from any major services.

File: specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json [hunk 5]
```diff
@@ -0,0 +1,905 @@
...
+    "application/json"
+  ],
+  "paths": {
+    "/providers/Microsoft.EdgeOrderPartner/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "description": "This method gets all the operations that are exposed for customer.",
+        "operationId": "ListOperationsPartner",
```
Comment (weidongxu-microsoft) [inline specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json:24]: Recommend to be `Operations_List`

**Review by weidongxu-microsoft — APPROVED** [review]
Leave some comment. Rest LGTM.

Let me know when you need it merged (recommend to be around the time your service online).

**Review by weidongxu-microsoft — COMMENTED** [review]
You would need a section of `swagger-to-sdk` in readme for SDK generation (unless you do not require any SDK or CLI).

E.g. compute again https://github.com/Azure/azure-rest-api-specs/tree/master/specification/compute/resource-manager#swagger-to-sdk

File: specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json [hunk 6]
```diff
@@ -0,0 +1,905 @@
...
+          "readOnly": true
+        },
+        "status": {
+          "description": "Billing status for the inventory",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "CloudError": {
```
Comment (weidongxu-microsoft) [inline specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json:300]: Why have `CloudError` when no one uses it? Please try clean up the unused schemas.

Comment (Pramila-dhaka) [reply specification/edgeorderpartner/resource-manager/Microsoft.EdgeOrderPartner/preview/2020-12-01-preview/edgeorderpartner.json:300]: We will take all these comments in next PR.

**Pull request merged**